From 6a178fd35fe259d90a4fd3265941b26df691b269 Mon Sep 17 00:00:00 2001
From: Thomas Tai <Thomas.Tai@windriver.com>
Date: Tue, 14 Oct 2008 16:37:05 -0400
Subject: [PATCH 3/3] net poll support for kgdboe

In order to use kgdb over ethernet, the ethernet driver must
support net polling api. A poll controller is added so that the
transmit and receive queue is being checked during kgdboe.

Signed-off-by: Thomas Tai <Thomas.Tai@windriver.com>
---
 drivers/net/tsi108_eth.c |   14 ++++++++++++++
 1 files changed, 14 insertions(+), 0 deletions(-)

diff --git a/drivers/net/tsi108_eth.c b/drivers/net/tsi108_eth.c
index 43fde99..e683b2c 100644
--- a/drivers/net/tsi108_eth.c
+++ b/drivers/net/tsi108_eth.c
@@ -1556,6 +1556,15 @@ static int tsi108_do_ioctl(struct net_device *dev, struct ifreq *rq, int cmd)
 	return generic_mii_ioctl(&data->mii_if, if_mii(rq), cmd, NULL);
 }
 
+#ifdef CONFIG_NET_POLL_CONTROLLER
+static void tsi108_poll_controller(struct net_device *dev)
+{
+	disable_irq(dev->irq);
+	tsi108_irq(dev->irq, dev);
+	enable_irq(dev->irq);
+}
+#endif
+
 static const struct ethtool_ops tsi108_ethtool_ops = {
 	.get_link 	= ethtool_op_get_link,
 	.get_settings	= tsi108_get_settings,
@@ -1625,6 +1634,11 @@ tsi108_init_one(struct platform_device *pdev)
 	dev->set_multicast_list = tsi108_set_rx_mode;
 	dev->get_stats = tsi108_get_stats;
 	netif_napi_add(dev, &data->napi, tsi108_poll, 64);
+
+#ifdef CONFIG_NET_POLL_CONTROLLER
+	dev->poll_controller = tsi108_poll_controller;
+#endif
+
 	dev->do_ioctl = tsi108_do_ioctl;
 	dev->ethtool_ops = &tsi108_ethtool_ops;
 
-- 
1.6.0

